### PR TITLE
Init refactor

### PR DESCRIPTION
**Pull request by gerbeldo**
# Description

R package testing infrastructure is only able to test files in the R folder (which should contain all source of an R package). This means that we were not able to test the `init.R` file and its functions. And this is a problem in particular for trying to test several pipeline steps in succession, given that they are tied together by the init file functions.

This PR refactors the init file, moving all function definitions to the `pipeline-runner/R/init-functions.R` file, renames some variables for consistency, adds Roxygen2 docs to all functions, moves some variables to the `sysdata.R` file (list of S3 bucket names), and removes some unnecessary `library` which are generally frowned upon.

The original init file was reduced to setup for running the pipeline, i.e. sourcing the functions, loading the constants and setting R options.

This PR will require Codecov checks to be disabled for merging. Codecov was so far ignoring the untested contents of the init file.

All contents of the man folder, as always can be ignored. They are automatically generated from the docstrings above functions.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-2145

#### Link to staging deployment URL (or set N/A)
https://ui-german-pipeline44c.scp-staging.biomage.net/

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [x] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/pipeline.code-workspace [hunk 1]
```diff
@@ -5,11 +5,11 @@
 		}
 	],
 	"settings": {
-        "r.rterm.linux": "/usr/local/bin/radian",
-        "r.bracketedPaste": true,
+		"r.bracketedPaste": false,
```
Comment (ivababukova) [inline pipeline-runner/pipeline.code-workspace:8]: can you explain the changes in this file?

File: pipeline-runner/R/init-functions.R [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+  states <- paws::sfn(config = pipeline_config$aws_config)
+  message("Loaded step function")
+
+  print(sessionInfo())
+
+  futile.logger::flog.layout(futile.logger::layout.simple)
+  futile.logger::flog.threshold(futile.logger::ERROR)
+
+  message("Waiting for tasks")
+
```
Comment (ivababukova) [inline pipeline-runner/R/init-functions.R:497]: are all these prints and messages useful:

```
message("Loaded pipeline config")
message("Loaded step function")
print(sessionInfo())
message("Waiting for tasks")
```
What is the difference between print and message?

File: pipeline-runner/R/init-functions.R [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+      futile.logger::appender.tee(file.path(dump_folder, "logs.txt"))
+    )
+
+    # start heartbeat as a different process in the background
+    message("Starting heartbeat")
+    # message inside r_bg will ONLY be printed into /tmp/[out|err]
+    # to see them
+    # 1. log into the R container
+    # 2. cat /tmp/out or tail -f /tmp/out
+    heartbeat_proc <- callr::r_bg(
```
Comment (ivababukova) [inline pipeline-runner/R/init-functions.R:526]: did you consider moving the heartbeat stuff into a separate function too?

**Review by ivababukova — COMMENTED** [review]
Really nice improvements to the codebase 😍 I added some small comments, I don't want to approve before checking that staging is working. Currently, the staged environment for this PR is down.

File: pipeline-runner/R/init-functions.R (hunk 3 above)
Comment (gerbeldo) [reply pipeline-runner/R/init-functions.R:526]: I considered it, yes, jaja. I'll do it

File: pipeline-runner/pipeline.code-workspace (hunk 1 above)
Comment (gerbeldo) [reply pipeline-runner/pipeline.code-workspace:8]: the tabsize for R files is 2 according to R style. It was not specified in this file so I added it.

Radian is a console for R written in python, and it has autocomplete and stuff. It's only meant for interactive development use, and not used in production (it's not even installed there, according to the dockerfile). 

I tried using it but has some compatibility issues with renv and could not get it to work properly. The default R console works just fine.

the bracketed paste is a parameter to accomodate running code blocks in the radian terminal.

These parameters can be set in vscode at the user level, if someone were to use radian, I am not sure anyone does.

File: pipeline-runner/R/init-functions.R (hunk 3 above)
Comment (gerbeldo) [reply pipeline-runner/R/init-functions.R]: done

File: pipeline-runner/pipeline.code-workspace (hunk 1 above)
Comment (ivababukova) [reply pipeline-runner/pipeline.code-workspace:8]: sounds good, if it doesn't work and it is not used, we should remove it.

**Review by ivababukova — APPROVED** [review]

**Pull request merged**